### DISCUSSION TITLE
Xcode 12 adding Push Notifications Capabilities

### DISCUSSION THREAD

**Discussion by heinrich-fresh** [ionic-team/capacitor#3687] [Q&A] [answered]
Hi, 

So I am implementing Push Notifications on an Ionic App (Angular) using Capacitor. 

From the documentation after you follow all the steps You have to add the Push Notification Capability manually in xcode under Signing And Capabilities tab.

But it looks like in Xcode 12 it looks different. 

This is what it use to look like:
![capacitor-xcode-capability](https://user-images.githubusercontent.com/58063979/96086883-42f9a280-0ec3-11eb-9064-27519789d0a3.png)

First Of all if you search of "Push" there is no capability
![Screenshot 2020-10-14 at 15 17 18](https://user-images.githubusercontent.com/58063979/96086948-5efd4400-0ec3-11eb-9ffd-340c2da396b0.png)

If you search for Notification there are 2 options:
![Screenshot 2020-10-14 at 15 17 25](https://user-images.githubusercontent.com/58063979/96086987-6fadba00-0ec3-11eb-80a7-7f95502b356b.png)

So which one is correct?

**Comment by jcesarmobile** [comment]
In my Xcode 12 I see exactly as in your first screenshot.

On the other ones look like you are creating a new target for push notifications, which is a totally different thing, that's used for sending push to Apple watch and things like that.

**Comment by ikeith** [comment, marked as answer]
As @jcesarmobile points out, it still looks like that. If the "Push Notifications" box is not appearing in that view below the signing section, that means you need to add it (it probably means your project doesn't have an entitlements file). Tap the "+ Capability" button that one of the arrows is pointing at and you will get a pop up with all of the possible entitlements listed so you can add "Push Notifications". Xcode will automatically create the entitlements file for you.
<img width="992" alt="entitlements" src="https://user-images.githubusercontent.com/1081183/96189379-5e3dcf80-0f0e-11eb-9093-92e974a97060.png">

**Comment by heinrich-fresh** [comment]
Awh really what a Noob. 

I was clicking on the + at the bottom instead of capability. 

Let's just say I had a long day.  

Thanks for pointing me in the right direction.

**Reply by ikeith** [reply]
To be fair, it is a confusing UI because the 'add' button at the top subverts convention. And Xcode certainly has a history of moving things around between versions. 😅 Glad you got it sorted it!
